src/server/task_manager/task_manager_mixin.ts

```diff
@@ -19,72 +19,34 @@
 
 import Joi from 'joi';
 import { TaskManagerClientWrapper } from './client_wrapper';
+import { getDefaultClient } from './default_client';
 import {
   TaskDefinition,
   TaskDictionary,
-  TaskManager,
   TaskManagerLogger,
-  TaskPool,
-  TaskStore,
   validateTaskDefinition,
 } from './task_pool';
-import { fillPool } from './task_pool/fill_pool';
-import { ConcreteTaskInstance, SanitizedTaskDefinition } from './task_pool/task';
-import { TaskPoller } from './task_pool/task_poller';
-import { TaskManagerRunner } from './task_pool/task_runner';
+import { SanitizedTaskDefinition } from './task_pool/task';
 
 export async function taskManagerMixin(kbnServer: any, server: any, config: any) {
-  const logger = new TaskManagerLogger((...args) => server.log(...args));
+  const logger = new TaskManagerLogger((...args: any[]) => server.log(...args));
   const totalCapacity = config.get('taskManager.num_workers');
   const definitions = extractTaskDefinitions(totalCapacity, kbnServer.uiExports.taskDefinitions);
 
-  server.decorate('server', 'taskManager', new TaskManagerClientWrapper());
+  server.decorate(
+    'server',
+    'taskManager',
+    new TaskManagerClientWrapper(logger, totalCapacity, definitions)
+  );
 
   kbnServer.afterPluginsInit(async () => {
-    const callCluster = server.plugins.elasticsearch.getCluster('admin').callWithInternalUser;
-    const store = new TaskStore({
-      index: config.get('taskManager.index'),
-      callCluster,
-      maxAttempts: config.get('taskManager.max_attempts'),
-      supportedTypes: Object.keys(definitions),
-    });
-
-    logger.debug('Initializing the task manager index');
-    await store.init();
-
-    const pool = new TaskPool({
-      logger,
-      totalCapacity,
-    });
-
-    const contextProvider = async (taskInstance: ConcreteTaskInstance) => ({
-      callCluster,
-      kbnServer,
-      taskInstance,
-    });
-
-    const poller = new TaskPoller({
-      logger,
-      pollInterval: config.get('taskManager.poll_interval'),
-      work: () =>
-        fillPool(
-          pool.run,
-          store.fetchAvailableTasks,
-          (instance: ConcreteTaskInstance) =>
-            new TaskManagerRunner({
```

Comment:
I moved this to the default client, and reordered the fields to be in the order that TaskManagerRunner defines. VS Code was giving a warning about this before & after I touched it